scripts/dumps/bulk_genome_dump_from_core.py

```diff
@@ -0,0 +1,334 @@
+            logging.error(
+                "Command %s exited %d: %s", " ".join(command), process.returncode, out
+            )
+            raise RuntimeError(
+                f"Command {' '.join(command)} exited {process.returncode}: {out}"
+            )
+
+        logging.info("Successfully ran: %s", " ".join(command))
+        return output.strip()
+
+
+def download_file(
+    host,
+    port,
+    core_db,
+    fasta_filename,
+    stdout_file="/dev/null",
+    stderr_file="/dev/null",
+    genome_component="",
+    mask="soft",
+):
+    """
+    Download the FASTA file from the core DB using a PERL script `dump_genome_from_core.pl`.
+
+    Args:
+        host (str): The DB host name.
+        port (str): The port number.
+        core_db (str): The core_db name.
+        fasta_filename (str): The name given for the FASTA file.
+        mask (str): The mask format for the FASTA file.
+
+    Returns:
+        str: The output of the subprocess call to download the FASTA file.
+    """
+    try:
+        work_dir = os.path.join(
+            os.environ["ENSEMBL_ROOT_DIR"],
+            "ensembl-compara",
+            "scripts",
+            "dumps",
+        )
+        script = "dump_genome_from_core.pl"
+
+        perl_call = [
+            "perl",
+            os.path.join(work_dir, script),
+            "--core_db",
+            core_db,
+            "--host",
+            host,
+            "--port",
+            str(port),
+            "--mask",
+            mask,
+            "-user",
+            "ensro",
+            "--outfile",
+            fasta_filename,
+        ]
+
+        # Conditionally add the genome component argument
+        if genome_component:
+            perl_call += ["--genome-component", genome_component]
+
+        logging.info("perl_call=%s", perl_call)
+        return subprocess_call(
+            command=perl_call,
+            use_job_scheduler=True,
+            job_name=f"{core_db}_{genome_component}",
+            stderr_file=stderr_file,
+            stdout_file=stdout_file,
+        )
+
+    except KeyError as e:
+        logging.error("Environment variable not set: %s", e)
+        raise
+
+    except Exception as e:
+        logging.error("An unexpected error occurred: %s", e)
+        raise
+
+
+def query_coredb(host, core_db, query):
+    """
+    Get the correct meta production name.
+
+    Args:
+        host (str): The DB host name.
+        core_db (str): The core_db name.
+        query (str): The query to be executed.
+
+    Returns:
+        str: The output of the subprocess call to query the database.
+    """
+    mysql_call = [host, core_db, "-N", "-e", query]
+    return subprocess_call(command=mysql_call)
+
+
+def parse_yaml(file, dest):
+    """
+    YAML parser.
+
+    Args:
+        file (file object): The file object.
+        dest (str): The destination PATH where the FASTA file will be stored.
+    """
+    content = yaml.load(file, Loader=SafeLoader)
```

Comment:
Could this line do with a little syntactic sugar? This would also simplify the YAML import, which could just be `import yaml`.
```suggestion
    content = yaml.safe_load(file)
```